core/store/src/adapter/trie_store.rs

```diff
@@ -100,18 +100,32 @@ impl<'a> TrieStoreUpdateAdapter<'a> {
         Self { store_update: StoreUpdateHolder::Reference(store_update) }
     }
 
+    /// Reads shard_uid mapping for given shard.
+    /// If the mapping does not exist, it means that `shard_uid` maps to itself.
+    fn read_shard_uid_mapping_from_db(
+        &self,
+        shard_uid: ShardUId,
+    ) -> Result<ShardUId, StorageError> {
+        let store = TrieStoreAdapter::new(Store::new(self.store_update.storage.clone()));
```

Comment:
We can just create a helper function here for accessing stuff from raw store and just use that instead?

Reply:
> What's the typical way to access data from store in here

From what I see elsewhere in the code we do not access data from within `StoreUpdate`.

> accessing stuff from raw store

Do you mean using `storage: Arc<dyn Database>` member of `StoreUpdate` directly?
`StoreUpdate` does not contain `Store`.

I would like to add `Store` to `TrieStoreUpdateAdapter`.
But it is problematic because of this:
```
impl<'a> TrieStoreUpdateAdapter<'a> {
    pub fn new(store_update: &'a mut StoreUpdate) -> Self {
        Self { store_update: StoreUpdateHolder::Reference(store_update) }
    }
```

The easiest way (changing 2 lines of code) would be to replace:
```
pub struct StoreUpdate {
    transaction: DBTransaction,
    storage: Arc<dyn Database>,
}
```
with
```
pub struct StoreUpdate {
    transaction: DBTransaction,
    store: Store,
}
```
where `Store` is just a wrapper for `storage`:
```
pub struct Store {
    storage: Arc<dyn Database>,
}
```

@wacban  @shreyan-gupta Wdyt?

Reply:
I applied the above change in 09a5ca75d1e09c6fa5dca62247273e175f5e277f so you can see it.